x/stake/keeper/query_utils.go

```diff
@@ -0,0 +1,55 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/stake/types"
+)
+
+// Return all validators that a delegator is bonded to. If maxRetrieve is supplied, the respective amount will be returned.
+func (k Keeper) GetDelegatorValidators(ctx sdk.Context, delegatorAddr sdk.AccAddress,
+	maxRetrieve ...int16) (validators []types.Validator) {
+
+	retrieve := len(maxRetrieve) > 0
```

Comment:
we shouldn't allow for retrieve all to exist - meaning that this retrieve term should be taken out - see my comments on the old PR